### ISSUE TITLE
Pattern matching failure on invalid use of higher-kinded type parameters

### ISSUE THREAD

**Issue by hakuch** [diml/utop#176]
I'm using `4.03.0+flambda` from opam and `utop` version `1.19.2`.

When one attempts to use a higher-kinded type parameter, the normal `ocaml` prompt gives a reasonable error:

```
$ ocaml
        OCaml version 4.03.0

# module type HIGHER = sig type _ t end ;;
module type HIGHER = sig type _ t end
# let foo (type k) (module H : HIGHER with type 'a t = 'a k) = 42 ;;
Error: invalid package type: parametrized types are not supported
```

When I attempt the same thing with `utop`:

```
$ utop

# module type HIGHER = sig type _ t end ;;
module type HIGHER = sig type _ t end
# let foo (type k) (module H : HIGHER with type 'a t = 'a k) = 42 ;;
Fatal error: exception File "src/lib/uTop.cppo.ml", line 246, characters 31-36: Pattern matching failed     
```

**Comment by lijunsong** [comment]
This one has been fixed in 1.19.3.

**Comment by hakuch** [comment]
I verified that this has been fixed as of `1.19.3`.
